uniffi/src/ffi/rustfuture.rs

```diff
@@ -0,0 +1,586 @@
+//!     call_status:: &mut ::uniffi::RustCallStatus,
+//! ) -> bool {
+//!     ::uniffi::ffi::uniffi_rustfuture_poll(future, waker, waker_environment, polled_result, call_status)
+//! }
+//! ```
+//!
+//! Let's analyse this function because it's an important one:
+//!
+//! * First off, this _poll FFI function_ forwards everything to
+//!   `uniffi_rustfuture_poll`. This later is generic, while the former has been
+//!   monomorphised by the procedural macros.
+//!
+//! * Second, it receives the `RustFuture` as an `Option<&mut RustFuture<_>>`. It
+//!   doesn't take ownership of the `RustFuture`! It borrows it (mutably). It's
+//!   wrapped inside an `Option` to check whether it's a null pointer or not.
+//!
+//! * Third, it receives a _waker_ as a pair of a _function pointer_ plus its
+//!   _environment_ if any: a null pointer is purposely allowed for the environment.
+//!   This waker function lives on the foreign language land. We will come back
+//!   to it in a second.
+//!
+//! * Fourth, it receives an in-out `polled_result` argument, that is filled with the
+//!   polled result if the future is ready (this part is subject to change).
+//!
+//! * Finally, the classical `call_status`, which is part of the calling API of `uniffi`.
+//!
+//! So, everytime this function is called, it polls the `RustFuture` after
+//! having reconstituted a valid [`Waker`] for it. As said earlier, we will come
+//! back to it.
+//!
+//! The last generated function is the _drop function_:
+//!
+//! ```rust,ignore
+//! #[no_mangle]
+//! pub extern "C" fn _uniffi_hello_drop(
+//!     future: Option<Box<::uniffi::RustFuture<bool>>>,
+//!     call_status: &mut ::uniffi::RustCallStatus
+//! ) {
+//!     ::uniffi::ffi::uniffi_rustfuture_drop(future, call_status)
+//! }
+//! ```
+//!
+//! First off, this _drop function_ is responsible to drop the `RustFuture`. It's
+//! clear by looking at its signature: It receives an `Option<Box<RustFuture<_>>>`,
+//! i.e. it takes ownership of the `RustFuture` _via_ `Box`!
+//!
+//! Similarly to the _poll function_, it forwards everything to
+//! `uniffi_rustfuture_drop`, which is the generic version of the monomorphised _drop
+//! function_.
+//!
+//! ## How `Future` works exactly?
+//!
+//! A [`Future`] in Rust does nothing. When calling an async function, it just
+//! returns a `Future` but nothing has happened yet. To start the computation,
+//! the future must be polled. It returns [`Poll::Ready(r)`][`Poll::Ready`] if
+//! the result is ready, [`Poll::Pending`] otherwise. `Poll::Pending` basically
+//! means:
+//!
+//! > Please, try to poll me later, maybe the result will be ready!
+//!
+//! This model is very different than what other languages do, but it can actually
+//! be mangled quite easily, hopefully for us!
+//!
+//! But… wait a minute… who is responsible to poll the `Future` if a `Future` does
+//! nothing? Well, it's _the executor_ (sometimes also called _the runtime_ but not
```

Comment:
At least in Rust terminology, the executor is only part of a runtime. But maybe not that important.

Reply:
Maybe I just remove this sentence, it could create confusion between the terminology of various foreign languages.